### PR TITLE
feat: home screen 

### PR DESCRIPTION
**Pull request by Fannolo**

### CODE REVIEW

File: whizzer/features/common/components/Carousel/Carousel.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+interface CarouselProps<T> extends ScrollViewProps {
+  data: Array<T>;
+  Component: React.FC<CarouselComponentType<T>>;
+}
+
+const Carousel = <T,>({ data, Component, ...rest }: CarouselProps<T>) => {
+  return (
+    <View>
+      <ScrollView horizontal {...rest}>
```
Comment (LRNZ09) [inline whizzer/features/common/components/Carousel/Carousel.tsx:12]: Use `FlatList` instead

File: whizzer/features/menu/components/CategoriesSlider/CategoriesSlider.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { View, Text } from "react-native";
+import React from "react";
+
+type Props = {};
```
Comment (LRNZ09) [inline whizzer/features/menu/components/CategoriesSlider/CategoriesSlider.tsx:4]: Needless type

**Review by LRNZ09 — CHANGES_REQUESTED** [review]

File: whizzer/features/home/components/RestaurantCard/RestaurantCard.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  };
+}
+
+const RestaurantCard = ({ item }: RestaurantCardProps) => {
+  const navigation = useNavigation();
+  return (
+    <View key={item.title} style={styles.container}>
+      <TouchableOpacity
+        onPress={() =>
+          navigation.navigate(RootScreenNames.RestaurantMenuScreen)
```
[the remark this answers is not in the document]
Comment (Fannolo) [reply whizzer/features/home/components/RestaurantCard/RestaurantCard.tsx:21]: never!

File: whizzer/features/common/components/Carousel/Carousel.tsx (hunk 1 above)
Comment (Fannolo) [reply whizzer/features/common/components/Carousel/Carousel.tsx:12]: Never!

File: whizzer/.env [hunk 4]
```diff
@@ -0,0 +1,2 @@
+SUPABASE_URL=https://abywcqquksfpaipjozbl.supabase.co
+SUPABASE_ANON_KEY=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImFieXdjcXF1a3NmcGFpcGpvemJsIiwicm9sZSI6ImFub24iLCJpYXQiOjE2NTY3NTM2MTIsImV4cCI6MTk3MjMyOTYxMn0.jvT4I8oW07U_geuoplLf1evIi45Xh5a8Jk-xCwirjXE
```
Comment (devteapot) [inline whizzer/.env:2]: Should't push this file

Comment (Fannolo) [reply whizzer/.env:2]: whooo says that

Comment (devteapot) [reply whizzer/.env:2]: god

File: whizzer/features/home/components/RestaurantCard/RestaurantCard.tsx [hunk 5]
```diff
@@ -2,33 +2,30 @@ import { View, Text, Image, StyleSheet, TouchableOpacity } from "react-native";
...
           <Image
             style={styles.image}
             source={require("../../../../assets/images/sushi_restaurant_cover.jpeg")}
             resizeMode={"cover"}
           />
           <View style={styles.textContainer}>
-            <Text>{item.title}</Text>
-            <Text>{item.distance}</Text>
+            <Text>{item.name}</Text>
+            <Text>69km</Text>
```
Comment (LRNZ09) [inline whizzer/features/home/components/RestaurantCard/RestaurantCard.tsx:28]: Should it show the `item.distance`?

File: whizzer/navigation/header/HomeScreenHeader/HomeScreenHeader.test.tsx [hunk 6]
```diff
@@ -0,0 +1,3 @@
+describe("HomeScreenHeader", () => {
+  it("should render the home screen header", () => {});
```
Comment (LRNZ09) [inline whizzer/navigation/header/HomeScreenHeader/HomeScreenHeader.test.tsx:2]: Add a real test or just use `it.todo`

File: whizzer/features/home/components/RestaurantCard/RestaurantCard.tsx [hunk 7]
```diff
@@ -2,33 +2,30 @@ import { View, Text, Image, StyleSheet, TouchableOpacity } from "react-native";
...
-        onPress={() =>
-          navigation.navigate(RootScreenNames.RestaurantMenuScreen)
-        }
-      >
+    <View key={item.id} style={styles.container}>
+      <TouchableOpacity onPress={handleNavigation}>
         <View style={styles.card}>
           <Image
             style={styles.image}
             source={require("../../../../assets/images/sushi_restaurant_cover.jpeg")}
```
[the remark this answers is not in the document]
Comment (Fannolo) [reply whizzer/features/home/components/RestaurantCard/RestaurantCard.tsx:23]: we are working on it in the restaurant menu MR

File: whizzer/features/home/components/RestaurantCard/RestaurantCard.tsx (hunk 5 above)
Comment (Fannolo) [reply whizzer/features/home/components/RestaurantCard/RestaurantCard.tsx:28]: working on the other MR

**Pull request merged**